File: 1704-determine-if-string-halves-are-alike/1704-determine-if-string-halves-are-alike.cpp
class Solution {
public:
    bool halvesAreAlike(string s) {
        unordered_set<char> st {'a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'};
        int a = 0;
        int b = 0;
        for (int i=0; i<s.length()/2; i++) {
            if (st.count(s[i])) a++;
        }
        for (int i=s.length()/2; i<s.length(); i++) {
            if (st.count(s[i])) b++;
        }
        return a == b;
    }
};

File: 1704-determine-if-string-halves-are-alike/1704-determine-if-string-halves-are-alike.py
class Solution:
    def halvesAreAlike(self, s: str) -> bool:
        x = 0
        n = len(s)
        v = "aeiouAEIOU"
        for c in s[:n//2]:
            if c in v:
                x += 1
        for c in s[n//2:]:
            if c in v:
                x -= 1
        return x == 0